src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -648,6 +650,89 @@ class ActorIdStringGenerator : public Generator<std::string> {
     std::string _actorId;
 };
 
+/** `{^Now: {}}` */
+class NowGenerator : public Generator<bsoncxx::types::b_date> {
+public:
+    NowGenerator(const Node& node, GeneratorArgs generatorArgs) {}
+    bsoncxx::types::b_date evaluate() override {
+        return bsoncxx::types::b_date{std::chrono::system_clock::now()};
+    }
+};
+
+// The 2 formats also cover "%Y-%m-%d", timezones are not parsed,
+// %z, %Q are output only https://www.boost.org/doc/libs/1_75_0/doc/html/date_time/date_time_io.html.
+const std::locale formats[] = {
+    std::locale(std::locale::classic(),new boost::posix_time::time_input_facet("%Y-%m-%dT%H:%M:%S.%F")),
+    std::locale(std::locale::classic(),new boost::posix_time::time_input_facet("%Y-%m-%d %H:%M:%S.%F")),
+};
+const size_t formats_n = sizeof(formats)/sizeof(formats[0]);
+const static boost::posix_time::ptime epoch(boost::gregorian::date(1970,1,1));
+
+static unsigned long long parseDateAsMillis(const std::string &datetime) {
```

Comment:
any reason why we use `unsigned long long` versus just `std::chrono::milliseconds`? I know the `std::chrono` API is a bit awkward but losing the precision/type has bitten me a couple times and here it would save you from having to do a `duration_cast` in at least one place I think. I've not used the boost chrono stuff you're using here, though, so maybe std::chrono isn't applicable.